### PR TITLE
CLOUD-579 - Fix upgrade test

### PR DESCRIPTION
**Pull request by tplavcic**
[CLOUD-579](https://jira.percona.com/browse/CLOUD-579)

This PR needs this: https://jira.percona.com/browse/K8SPSMDB-289
otherwise the test fails.

**EDIT:**
This is now waiting for this task: https://jira.percona.com/browse/K8SPSMDB-293

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade | failed |
| monitoring | passed |
| upgrade-consistency | passed |
| limits | passed |
| scaling | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| security-context | passed |
| arbiter | passed |
| scheduled-backup | passed |
| self-healing | passed |
| service-per-pod | passed |
| operator-self-healing | passed |
| liveness | passed |
| users | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| storage | passed |
| upgrade | failed |
| self-healing | passed |
| monitoring | passed |
| limits | passed |
| upgrade-consistency | passed |
| scaling | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| arbiter | passed |
| scheduled-backup | passed |
| service-per-pod | passed |
| liveness | passed |
| users | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade | failed |
| self-healing | passed |
| monitoring | passed |
| limits | passed |
| scaling | passed |
| monitoring-2-0 | passed |
| upgrade-consistency | passed |
| security-context | passed |
| operator-self-healing | passed |
| demand-backup | passed |
| arbiter | passed |
| scheduled-backup | passed |
| service-per-pod | passed |
| liveness | passed |
| users | passed |

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-442-03e98f41

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| one-pod | passed |
| init-deploy | passed |
| upgrade | passed |
| self-healing | passed |
| monitoring | passed |
| limits | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scaling | passed |
| monitoring-2-0 | passed |
| operator-self-healing | passed |
| scheduled-backup | passed |
| security-context | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |

**Review by cap1984 — APPROVED** [review]

File: e2e-tests/upgrade/run [hunk 1]
```diff
@@ -8,117 +8,182 @@ test_dir=$(realpath $(dirname $0))
...
+                exit 1
+            fi
+            ;;
+    esac
+}
+
+function deploy_operator_gh() {
+    desc 'start operator'
+    curl -s "https://raw.githubusercontent.com/percona/percona-server-mongodb-operator/${GIT_TAG}/deploy/crd.yaml" > "${tmp_dir}/crd_${GIT_TAG}.yaml"
+    kubectl_bin apply -f "${tmp_dir}/crd_${GIT_TAG}.yaml"
```
Comment (hors) [inline e2e-tests/upgrade/run:96]: kubectl_bin apply -f "https://raw.githubusercontent.com/percona/percona-server-mongodb-operator/${GIT_TAG}/deploy/crd.yaml"

File: e2e-tests/upgrade/run [hunk 2]
```diff
@@ -8,117 +8,182 @@ test_dir=$(realpath $(dirname $0))
...
-    destroy $namespace
+    check_applied_images "all"
+    compare_generation "2" "statefulset" "${CLUSTER}-rs0"
+    compare_generation "2" "psmdb" "${CLUSTER}"
+
+    desc 'cleanup'
+    kubectl_bin delete -f "${cr_yaml}" || true
+    kubectl_bin delete -f "${src_dir}/deploy/crd.yaml" || true
+    kubectl_bin delete pvc --all
+    kubectl_bin delete -f "${src_dir}/deploy/rbac.yaml" || true
```
Comment (hors) [inline e2e-tests/upgrade/run:185]: Why do you need do delete all these stuff manually? Maybe it is better to just use  `destroy "${namespace}"` or improve  destroy function if it is needed.

**Review by hors — CHANGES_REQUESTED** [review]
please fix conflicts

File: e2e-tests/upgrade/run (hunk 2 above)
Comment (tplavcic) [reply e2e-tests/upgrade/run]: moved to destroy function

File: e2e-tests/upgrade/run (hunk 1 above)
Comment (tplavcic) [reply e2e-tests/upgrade/run]: done

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| init-deploy | passed |
| one-pod | failed |
| upgrade | passed |
| self-healing | passed |
| upgrade-consistency | passed |
| limits | passed |
| demand-backup | failed |
| scaling | passed |
| operator-self-healing | passed |
| security-context | passed |
| monitoring | failed |
| scheduled-backup | failed |
| monitoring-2-0 | passed |
| arbiter | failed |
| service-per-pod | failed |
| liveness | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| upgrade | passed |
| self-healing | passed |
| limits | passed |
| upgrade-consistency | passed |
| scaling | passed |
| demand-backup | failed |
| security-context | passed |
| operator-self-healing | passed |

**Review by cap1984 — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade | failed |
| self-healing | passed |
| limits | passed |
| scaling | passed |
| upgrade-consistency | passed |
| operator-self-healing | passed |
| demand-backup | failed |
| security-context | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | failed |
| liveness | passed |
| smart-update | passed |
| scheduled-backup | failed |
| version-service | passed |
| users | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | failed |
| upgrade | failed |
| storage | passed |
| self-healing | passed |
| limits | passed |
| upgrade-consistency | passed |
| scaling | passed |
| demand-backup | failed |
| operator-self-healing | passed |
| security-context | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade | failed |
| self-healing | passed |
| monitoring | passed |
| limits | passed |
| scaling | passed |
| monitoring-2-0 | passed |
| upgrade-consistency | passed |
| security-context | passed |
| demand-backup | passed |
| operator-self-healing | passed |
| arbiter | passed |
| service-per-pod | passed |
| scheduled-backup | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | failed |

**Review by hors — APPROVED** [review]

**Pull request merged**